napari/_qt/__init__.py

```diff
@@ -27,6 +27,17 @@
         Path(PySide2.__file__).parent / 'Qt' / 'plugins'
     )
 
+if API_NAME == 'PySide6' and sys.version_info[:2] < (3, 10):
+    from packaging import version
+
+    if version.parse(QT_VERSION) > version.parse("6.3.1"):
+        raise RuntimeError(
```

Comment:
I created an environment on my Intel mac with Python 3.9.13 and PySide6==6.4.0 (see [output of `pip list`](https://gist.github.com/andy-sweet/0269f3cfafad49a6fe0f41f8b4d4e984) for other details).

But when I run napari, I get the [enum error message related to the PySide6 issue](https://gist.github.com/andy-sweet/f7b3db105b233e00854726d37b04aa4c).

I was expecting to get this error first. Or at least an error from qtpy until a new version is released. Did I misunderstand what this PR is doing?

Reply:
I guess we're getting the enum-error as we try to show the original error?

Reply:
I just found out how many places we import qtpy directly...

Reply:
Ok, I fix the code by proper move qt import. 

The PySide6 was imported in `magicgui` that means that our wold fixes in `napari._qt` may also not work.

Thanks for finding this before merging.